### ISSUE TITLE
June 2020 PythonPune Newsletter Collection Box

### ISSUE THREAD

**Issue by raukadah** [pythonpune/meetup-talks#101]
Let's use this issue to collect python related news, new package updates and events happening around Pune. We can compile and publish it during **June PythonPune** meetup. 

## How to add news:
Comment on this issue with news link and news topic and also include the source from we got this. 

## How to add new package release:
Paste the link of new package release

## How to include events here:
If a new event is happening in Pune and wants to share in the newsletter, feel free to drop the event link with date and venue.

**Comment by girish946** [comment]
Project: snakeware -  <https://github.com/joshiemoore/snakeware>

snakeware is a free Linux distro with a Python userspace inspired by the Commodore 64. You are booted directly into a Python interpreter, which you can use to do whatever you want with your computer.

**Comment by answerquest** [comment]
Found a useful function to flatten nested dicts from real-world jsons with multiple depths and arrays mixed in:  
https://github.com/ScriptSmith/socialreaper/blob/master/socialreaper/tools.py#L8

Cross-posted here: https://stackoverflow.com/a/62186053/4355695

**Comment by amolkahat** [comment]
Curl Cookbook: https://catonmat.net/cookbooks/curl

**Comment by Pradhvan** [comment]
Pattern Matching, Draft for PEP: 622  https://github.com/gvanrossum/patma

A match statement takes an expression and compares it to successive patterns given as one or more case blocks.  

**tl;dr** `match` statement which is similar to `switch` statements might be added to Python if **PEP 622** is accepted. 

```python
def http_error(status):
    match status:
        case 400:
            return "Bad request"
        case 401:
            return "Unauthorized"
        case 403:
            return "Forbidden"
        case 404:
            return "Not found"
        case 418:
            return "I'm a teapot"
        case _:
            return "Something else"
```

**Comment by raukadah** [comment]
PyLadies India embarked its journey - https://anweshadas.in/pyladies-india-embarked-its-journey/
tldr:
* first unified meetup
* Anwesha shared her journey with PyLadies India
* Marriata had given a nice workshop on creating a github bot

https://nedbatchelder.com/blog/202006/pickles_nine_flaws.html
Pickle’s nine flaws
tldr:
* pickle - serialize and de-serialize objects
* Not to use pickle due to these flaws i.e. insecure, Over-serializes, unreadable

**Comment by raukadah** [comment]
https://pyfound.blogspot.com/2020/06/2020-python-software-foundation-board_26.html
2020 Python Software Foundation Board of Directors Election Retrospective and Next Steps
tldr:
* Full results and participation & representation
* Further steps to improve the demographics of our members and voters

**Comment by raukadah** [comment]
https://myst-parser.readthedocs.io/en/latest/index.html
MyST - Markedly Structured Text
tldr:
* A fully-functional markdown flavor and parser for Sphinx just like rst for readthedocs
* CommonMark markdown (such as Jupyter Notebook markdown) is natively supported by the MyST parser.

 https://docs.readthedocs.io/en/latest/guides/autobuild-docs-for-pull-requests.html
Autobuild Documentation for Pull Requests
tldr:
* build docs on each pull request via a warning banner
* Get build status notification
* few troubleshooting steps if you hit issues

**Comment by raukadah** [comment]
https://quansight.github.io/ndindex/slices.html
* Slices
tldr:
* slicing is too confusting, let's read it and make it clear
* Where we are confused while using it and how to fix that

**Comment by raukadah** [comment]
https://snarky.ca/what-is-the-core-of-the-python-programming-language/
What is the core of the Python programming language?
tldr:
* A question started with usage of python with webassembly?
* Does a python needs REPL?
* How much compatibility is necessary to be useful with existing softwares

**Comment by raukadah** [comment]
New projects to try:
* https://github.com/xxh/xxh -> Bring your favorite shell wherever you go through the ssh
* https://github.com/adamian98/pulse -> Self-Supervised Photo Upsampling via Latent Space Exploration of Generative Models
* https://pypi.org/project/plot-coverage/ -> Python package to generate plots from coverage report

**Comment by raukadah** [comment]
https://foss.training/
DGPLUG SUMMER TRAINING starting on 30th Jun at 13:30 UTC

**Comment by Pradhvan** [comment]
@raukadah I think we should close this now as #104 is merged.  😄
